### DISCUSSION TITLE
Unable to Activate Copilot in VS2022

### DISCUSSION THREAD

**Discussion by arosey** [community/community#32472] [Copilot Conversations] [answered]
The dialog says, "A device activation code will be copied to the clipboard..." but when I click ok, nothing is copied to my clipboard.
Visual Studio Professional 2022 System Info:

Microsoft Visual Studio Professional 2022
Version 17.3.3
VisualStudio.17.Release/17.3.3+32825.248
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Professional

Visual C++ 2022   00476-80000-00000-AA056
Microsoft Visual C++ 2022

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

ASA Service Provider   1.0

ASP.NET and Web Tools   17.3.376.3011
ASP.NET and Web Tools

ASP.NET Web Frameworks and Tools 2012   17.3.376.3011
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   17.3.376.3011
Azure App Service Tools v3.0.0

Azure Data Lake Tools for Visual Studio   2.6.5000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   17.3.376.3011
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.6.5000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   4.3.0-3.22423.10+b35bb0baca1071d50cc4f8cf5a1a89e7cc112461
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Cookiecutter   17.0.22179.3
Provides tools for finding, instantiating and customizing templates in cookiecutter format.

GitHub Copilot   1.43.0.1 (v1.43.0.1@cb116db27)
GitHub Copilot is an AI pair programmer that helps you write code faster and with less work.

GitHub Copilot Agent   1.43.6621 (v1.43.0)

Microsoft Azure Hive Query Language Service   2.6.5000.0
Language service for Hive query

Microsoft Azure Stream Analytics Language Service   2.6.5000.0
Language service for Azure Stream Analytics

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

NuGet Package Manager   6.3.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Python - Profiling support   17.0.22179.3
Profiling support for Python projects.

Python with Pylance   17.0.22179.3
Provides IntelliSense, projects, templates, debugging, interactive windows, and other support for Python developers.

Razor (ASP.NET Core)   17.0.0.2232702+e1d654e792aa2fe6646a6935bcca80ff0aff4387
Provides languages services for ASP.NET Core Razor.

SQL Server Data Tools   17.0.62207.04100
Microsoft SQL Server Data Tools

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   17.0.10701.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.3.0-3.22423.10+b35bb0baca1071d50cc4f8cf5a1a89e7cc112461
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.22363.4+1b94f89d4d1f41f20f9be73c76f4b229d4e49078
Microsoft Visual F# Tools

Visual Studio Inline Suggestions   2.2.2055.12395
Ghost text API for Visual Studio inline suggestions

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

**Comment by iamcjwilliams** [comment, marked as answer]
The engineering team is currently investigating a regression with VS 17.3.x, if you can please try using Copilot 1.43.0.1 with VS 17.2.x which should not cause completions to hang after a while. Thanks!

**Reply by arosey** [reply]
Thanks for replying!
It is working now. I updated both Visual Studio and Copilot and now the code is copied to my clipboard as expected.
Version 17.3.4
VisualStudio.17.Release/17.3.4+32901.215
GitHub Copilot 1.44.0.0 (v1.44.0.0@47e914df6)
GitHub Copilot Agent 1.44.6735 (v1.44.0)

**Reply by ChrisPorter87** [reply]
I'm having a similar issue.  When you clicked the ok in the box that pops up telling about the code being saved to the clipboard, did it immediately redirect you to the site? When I click ok, nothing happens and I can't figure out where to paste this supposed code to.  Thank you

**Reply by ChrisPorter87** [reply]
How did you adjust the copilot version number?

**Reply by arosey** [reply]
That is the same issue I was having. When I clicked ok it did not open up a webpage nor did it copy a code to my clipboard (If it is copying a code to your clipboard, simply paste it here: https://github.com/login/device).
To adjust the copilot version number, I had to wait for an update to appear in Visual Studio > Extensions > Updates.

**Reply by dustindodson** [reply]
So there's no way to get the "code" that's supposed to be copied to the clipboard?

And there's no way to adjust the version of copilot until there is a new version made once every 8 months?

Great.... there goes 100 bucks.

oh yeah, uninstalling it and re-installing it does nothing as well.

**Reply by bhushanvinay** [reply]
same nothing copied to clipboard.

**Reply by ayim** [reply]
@bhushanvinay the authentication scheme for the GitHub Copilot extension in VS has changed.

1) Make sure "Sign in with device flow" is set to false
<img width="523" alt="image" src="https://github.com/community/community/assets/10951869/7e22d203-071b-4924-8e4b-f84c2e3e1a3a">

2) Add your GitHub Account (make sure it has Copilot activated) to Visual Studio
<img width="518" alt="image" src="https://github.com/community/community/assets/10951869/3c1b0f7b-2452-4dc9-af43-bf551451d3b4">

**Comment by diva-21** [comment]
Do check the latest version of VS code and login once again
for me it has worked
